L1Trigger/L1TMuonEndCap/src/PtAssignmentEngineDxy.cc

```diff
@@ -101,140 +89,93 @@ void PtAssignmentEngineDxy::preprocessing_dxy(const EMTFTrack& track, emtf::Feat
 
   // Get valid pattern values
   if (st1)
-    pat1 = data.cpattern[0];
+    x_csc_pattern[0] = data.cpattern[0];
   if (st2)
-    pat2 = data.cpattern[1];
+    x_csc_pattern[1] = data.cpattern[1];
   if (st3)
-    pat3 = data.cpattern[2];
+    x_csc_pattern[2] = data.cpattern[2];
   if (st4)
-    pat4 = data.cpattern[3];
-
-  // F/R bit
-  fr_1 = data.fr[0];
-
-  // RPC hit in station
-  rpc_1 = (st1 ? (pat1 == 0) : 0);
-  rpc_2 = (st2 ? (pat2 == 0) : 0);
-  rpc_3 = (st3 ? (pat3 == 0) : 0);
-  rpc_4 = (st4 ? (pat4 == 0) : 0);
-
-  // Calculate bends from patterns
-  bend_1 = aux().calcBendFromPattern(pat1, track.Endcap());
-  bend_2 = aux().calcBendFromPattern(pat2, track.Endcap());
-  bend_3 = aux().calcBendFromPattern(pat3, track.Endcap());
-  bend_4 = aux().calcBendFromPattern(pat4, track.Endcap());
-
-  // Invalid bend value is 0 in the NN
-  if (bend_1 == -99)
-    bend_1 = 0;
-  if (bend_2 == -99)
-    bend_2 = 0;
-  if (bend_3 == -99)
-    bend_3 = 0;
-  if (bend_4 == -99)
-    bend_4 = 0;
-
-  // In the emulator RPCs get assigned abs(bend) = 5. This needs to be 0 for the NN.
-  if (std::abs(bend_1) == 5 && rpc_1 == 1)
-    bend_1 = 0;
-  if (std::abs(bend_2) == 5 && rpc_2 == 1)
-    bend_2 = 0;
-  if (std::abs(bend_3) == 5 && rpc_3 == 1)
-    bend_3 = 0;
-  if (std::abs(bend_4) == 5 && rpc_4 == 1)
-    bend_4 = 0;
+    x_csc_pattern[3] = data.cpattern[3];
 
   // Calculate delta phi
-  dPhi_12 = (data.delta_ph[0] != invalid_dphi) ? data.delta_ph[0] * (data.sign_ph[0] ? 1 : -1) : 0;
-  dPhi_13 = (data.delta_ph[1] != invalid_dphi) ? data.delta_ph[1] * (data.sign_ph[1] ? 1 : -1) : 0;
-  dPhi_14 = (data.delta_ph[2] != invalid_dphi) ? data.delta_ph[2] * (data.sign_ph[2] ? 1 : -1) : 0;
-  dPhi_23 = (data.delta_ph[3] != invalid_dphi) ? data.delta_ph[3] * (data.sign_ph[3] ? 1 : -1) : 0;
-  dPhi_24 = (data.delta_ph[4] != invalid_dphi) ? data.delta_ph[4] * (data.sign_ph[4] ? 1 : -1) : 0;
-  dPhi_34 = (data.delta_ph[5] != invalid_dphi) ? data.delta_ph[5] * (data.sign_ph[5] ? 1 : -1) : 0;
+  x_dphi[0] = (data.delta_ph[0] != invalid_dphi) ? data.delta_ph[0] : 0;
```

Comment:
A `for` loop would simplify the code now that you have indexed also the name of the variables